### ISSUE TITLE
ZIP/gzip the word lists

### ISSUE THREAD

**Issue by hauleth** [m1dnight/horsestaplebattery#1]
Erlang have built-in support for popular compression formats:

- http://www.erlang.org/doc/man/zlib.html
- http://www.erlang.org/doc/man/zip.html

This should slightly reduce the library size.

**Comment by arjan** [comment]
This only reduces the dev-time download; the word list is assembled in compile time so zipping the txt files does not reduce the runtime library size.

**Comment by hauleth** [comment]
@arjan that would be second optimisation: store all data in memory via `GenServer` or `Agent` started with application.

**Comment by arjan** [comment]
Yes, or even better in an ETS table. This is exactly the phase that the `:tzdata` project also went through.. first compiled in and then moved into ETS. However the TZ database is bigger, I guess.

Alternatively you could create a public API to serve these names ;-)

**Comment by m1dnight** [comment]
8 years later, and I can close this. Talk about velocity!

I ended up changing something else too: i read the files at compiletime and write them to a bin file as binary terms. Then I load then lazily into persistent_term storage.
